### ISSUE TITLE
[SLP] Vectorizer runs forever after PR #88504

### ISSUE THREAD

**Issue by chsigg** [llvm/llvm-project#91467]
After 37ae4ad0eef338776c7e2cffb3896153d43dcd90, the `vectorize-slp` pass does not complete on  [this IR](https://github.com/llvm/llvm-project/files/15248904/repro.ll.txt) produced by Triton with

```
LLVM_IR_ENABLE_DUMP=1 pytest --verbose --capture=no \
  python/test/unit/hopper/test_persistent_warp_specialized_gemm.py \
  -k '128-128-128-4-2-360-4096-512-False-True-none-float32-True-3'
```
This blocks upgrading Triton to the latest LLVM version (https://github.com/openai/triton/pull/3818).

I'm not familiar with the vectorizer code and it's not obvious to me what's going wrong. I would therefore highly appreciate any help looking into the issue.

**Comment by RKSimon** [comment]
CC @alexey-bataev

**Comment by alexey-bataev** [comment]
Fixed in 2475efa91d8b4fa8f1a2d16052cb6d14be7d5dc6

**Comment by chsigg** [comment]
Thank you very much Alexey for the super quick fix!
